components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/EndpointCertificatesApiServiceImpl.java

```diff
@@ -226,6 +233,61 @@ public Response updateEndpointCertificateByAlias(String alias, InputStream certi
         return null;
     }
 
+    @Override
+    public Response getCertificateUsageByAlias(String alias, Integer limit, Integer offset, MessageContext messageContext) throws APIManagementException {
+
+        List<API> allMatchedApis = new ArrayList<>();
+        Object apiListDTO;
+
+        limit = limit != null ? limit : RestApiConstants.PAGINATION_LIMIT_DEFAULT;
+        offset = offset != null ? offset : RestApiConstants.PAGINATION_OFFSET_DEFAULT;
+
+        String userName = RestApiCommonUtil.getLoggedInUsername();
+        CertificateMetadataDTO certificateMetadataDTO;
+
+        try {
+
+            if (StringUtils.isEmpty(alias)) {
+                RestApiUtil.handleBadRequest("The alias should not be empty", log);
+            }
+
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            certificateMetadataDTO = apiProvider.getCertificate(userName, alias);
+            String endpoint = certificateMetadataDTO.getEndpoint();
+
+            if(endpoint.contains("://")){
```

Comment:
Add a constant with applicable name for this

Reply:
done